ert/data/record/_record.py

```diff
@@ -190,44 +190,51 @@ class RecordCollectionType(str, Enum):
 
 
 class RecordCollection:
+    """Storage container for multiple records of the same type.
+
+    If the collection type is uniform, all records are identical. The typical use case
+    for a uniform collection is a record that is constant over an ensemble.
+    """
+
     def __init__(
         self,
         records: Tuple[Record, ...],
-        ensemble_size: Optional[PositiveInt] = None,
+        length: Optional[PositiveInt] = None,
```

Comment:
Using `__len__` suggests a container, which can be empty, but supposedly still exists. By allowing the length to be `None`,  it becomes a bit weird: the collection can be in some invalid state. I do not remember what the use case was for allowing the `ensemble_size` field to be `None`, but maybe we should look if this is really necessary. Having `len(collection)` return `None`, or indeed anything else than an integer is unexpected.

Reply:
This `None`ness is only a thing when calling `__init__`, after the object is intialized, `len(..)` cannot return `None`. The question still remains; why can't we have an empty `RecordCollection` (as it will give a `ValueError` if attempted)?

Reply:
Allowing `None` in `__init__` seems fair enough, as it can be inferred from the tuple-length when not provided, so it is more of a convenience thing.

Reply:
Ah, yes, if the length is always set, its not a problem.

The `ValueError` is probably a remainder form the original intention that it would also be an ensemble, which should not be empty. If we want a generic collection type, that test maybe should go, but then in other places in the code additional checks may be needed, maybe not very practical.